extensions/rocksdb-repos/RocksDbStream.h

```diff
@@ -160,7 +160,7 @@ class RocksDbStream : public io::BaseStream {
 
   std::string value_;
 
-  rocksdb::DB *db_;
+  gsl::not_null<minifi::internal::RocksDatabase*> db_;
```

Comment:
I was under the impression that taking a reference indicates that we are not planning on storing it, is this not a generally recognized convention?

Reply:
I was not aware of this convention; to me, a reference member variable sounds like a clear expression of "this object depends on `db_` being non-null and should not outlive it".  I would be interested to hear what others think.

Reply:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#note-392 Last sentence of "Exceptions":
> Note that using a reference member is almost always wrong.
Sadly there is no explanation and I couldn't find one elsewhere either.

I was not aware of the convention either, that taking a reference indicates not storing it. Do you have any resources explaining it? My disk space watchdog PR is affected because it takes ctor args by reference just to store them in a `gsl::not_null<T*>` member.

Reply:
also related: https://abseil.io/tips/177

Reply:
I couldn't find a convincing argument either, only a gut feeling that having `U f(T& t)` and `U g(T* t)`, after calling `g` I'm more reluctant to discard the `t` I provided than after calling `f`, I feel like if I write `f` I'm signaling that I only need the object for the duration of the function

Reply:
but since nobody seems to share this feeling (here or the "internet") I will modify the constructor to take a reference 👍